### DISCUSSION TITLE
How do I use the packet driver?

### DISCUSSION THREAD

**Discussion by mikesixes** [dosemu2/dosemu2#2106] [Q&A]
To use my packet driver programs in DOS, I invoke the packet driver with its name and interrupt number, ie "getpkt 0x60". Do I do this in Dosemu2 as well, and if so, what is the name of the Dosemu2 packet driver?
 My application is not a network application, it just uses ethernet as a fast serial connection. "eth" would seem like the natural choice for that, but it's deprecated. Any advice on what the best choice would be appreciated. Tap seems like a good choice, but I don't know what I'm talking about. It sends small packets fairly frequently, every few milliseconds.

**Comment by stsp** [comment]
Packet driver always has the name
"PKT DRVR" on interrupt entry header.
Its quite difficult to understand what
are you talking about.

**Reply by mikesixes** [reply]
Thank you for your reply. I have previously been running my app on a board that boots DOS. In DOS, you have to load a packet driver before you can use it. For instance, my autoexec.bat file would have a line that says "fetpkt 0x60" if the packet driver was called fetpkt.com. I was simply wondering if I needed to load the driver, and if so what it was called. It later occurred to me that I could learn something by trying to load one of my previous packet drivers, and when I did that, it gave me the message that there's already a packet driver at interrupt 0x60, so I now know that the packet driver is apparently always available in dosemu2. I haven't got it working yet, but at least I know it's there. I very much appreciate you taking the time to reply to my question. I'm sure I'll have others as I get further into this. Do you think tap would be the best choice for a non-networking ethernet app? It just sends and receives raw ethernet packets between 2 computers connected by a crossover cable. The DOS computer receives sensor data from a microcontroller-based board and sends actuator signals to it. Both computers just send to the broadcast address so they don't need to know each other's MAC address and of course they don't need any IP address since there is no network involved. Thanks again for your time.

**Reply by stsp** [reply]
"eth" rather than "tap" is the way
to occupy the NIC exclusively.
Which seems to be what you are
looking for.

**Reply by mikesixes** [reply]
Thank you sir!

**Comment by andrewbird** [comment]
Here's the info reported by the mTCP pkttool when querying the built-in pkt driver, perhaps that's what you are looking for?
~~~
C:\>pkttool scan

mTCP pkttool by M Brutman (mbbrutman@gmail.com) (C)opyright 2012-2022
Version: Jul  1 2022

Scanning:

Details for driver at software interrupt: 0x60
  Name: Linux$
  Entry point: F000:F57A
  Version: 1   Class: 1   Type: 12  Interface Number: 1
  Function flag: 2  (basic and extended functions)
  Current receive mode: all packets (promiscuous mode)
  MAC address: 66:62:78:A6:0B:16
~~~

**Reply by mikesixes** [reply]
Thank you! You've anticipated one of my other questions about the driver.
